drivers/adc/Kconfig.stm32f4

```diff
@@ -0,0 +1,373 @@
+# Kconfig - ADC configuration options
+
+#
+# Copyright (c) 2015 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ADC_STM32F4
+	bool "STM32F4 ADC driver"
+	depends on SOC_FAMILY_STM32
+	select USE_STM32_HAL_GPIO
+	select USE_STM32_HAL_GPIO_EX
+	select USE_STM32_HAL_ADC
+	select USE_STM32_HAL_ADC_EX
+	select USE_STM32_HAL_CORTEX
+	help
+		Enable the driver implementation for the stm32f4xx ADC
+
+if ADC_STM32F4
+
+	menu ADC_0
+		config ADC0_CHAN0
+			bool "channel 0 (PA0)"
+			default n
+			depends on ADC_0
+			help
+			  Enables ADC channel 0 at pin PA0
```

Comment:
maybe we can do the configuration of the driver instance in the dts?
Think configuration in run time is a waste of resources. We do not need to check the config before every adc conversion. It can be checked at the init of the driver once(and then never again)

Reply:
DTS is only to describe the hardware, not the configuration.

I agree that configuring at runtime will have a cost, but it is way more flexible. Static configuration may be OK for a simple project, but it is not enough for a more complex usecase. Think for example that you may have to reconfigure your ADC at several points in the lifetime of your product's lifecycle (i.e. production/testing, shipping, everyday use, low-power saving mode, etc.), which is not possible with the current implementation.

I do not remember if the new ADC API is better with respect to this point.